core/audits/layout-shifts.js

```diff
@@ -0,0 +1,148 @@
+
+const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
+
+class LayoutShifts extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'layout-shifts',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      scoreDisplayMode: Audit.SCORING_MODES.METRIC_SAVINGS,
+      guidanceLevel: 2,
+      requiredArtifacts: ['traces', 'TraceEngineResult', 'TraceElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const clusters = artifacts.TraceEngineResult.data.LayoutShifts.clusters ?? [];
+    const {cumulativeLayoutShift: clsSavings, impactByNodeId} =
+      await CumulativeLayoutShiftComputed.request(trace, context);
+
+    /** @type {Item[]} */
+    const items = [];
+    const layoutShiftEvents = clusters.flatMap(c => c.events);
+    for (const event of layoutShiftEvents) {
+      const biggestImpactNodeId = TraceElements.getBiggestImpactNodeForShiftEvent(
+        event.args.data.impacted_nodes, impactByNodeId);
+      const biggestImpactElement = artifacts.TraceElements.find(
+        t => t.traceEventType === 'layout-shift' && t.nodeId === biggestImpactNodeId);
+
+      // Turn root causes into sub-items.
+      const index = layoutShiftEvents.indexOf(event);
+      const rootCauses = artifacts.TraceEngineResult.rootCauses.layoutShifts[index];
+      /** @type {SubItem[]} */
+      const subItems = [];
+      if (rootCauses) {
+        for (const cause of rootCauses.fontChanges) {
+          const url = cause.request.args.data.url;
+          subItems.push({
+            url,
+            cause: str_(UIStrings.rootCauseFontChanges),
+          });
+        }
+        for (const cause of rootCauses.iframes) {
+          const element = artifacts.TraceElements.find(
+            t => t.traceEventType === 'layout-shift' && t.nodeId === cause.iframe.backendNodeId);
+          subItems.push({
+            node: element ? Audit.makeNodeItem(element.node) : undefined,
+            cause: str_(UIStrings.rootCauseInjectedIframe),
+          });
+        }
+        for (const cause of rootCauses.renderBlockingRequests) {
+          const url = cause.request.args.data.url;
+          subItems.push({
+            url,
+            cause: str_(UIStrings.rootCauseRenderBlockingRequest),
+          });
+        }
+        for (const cause of rootCauses.unsizedMedia) {
+          const element = artifacts.TraceElements.find(
+            t => t.traceEventType === 'layout-shift' && t.nodeId === cause.node.backendNodeId);
+          subItems.push({
+            node: element ? Audit.makeNodeItem(element.node) : undefined,
+            cause: str_(UIStrings.rootCauseUnsizedMedia),
+          });
+        }
+      }
+
+      items.push({
```

Comment:
In some scenarios this audit is empty even though there were layout shifts ([example](https://googlechrome.github.io/lighthouse/viewer/?gist=b97da9762d7e657b97b9e9564609bcb1))